presto-main/src/main/java/com/facebook/presto/connector/ConnectorManager.java

```diff
@@ -482,6 +493,15 @@ public MaterializedConnector(ConnectorId connectorId, Connector connector)
             }
             this.metadataUpdaterProvider = Optional.ofNullable(metadataUpdaterProvider);
 
+            ConnectorMetadataUpdateHandleSerdeProvider metadataUpdateHandleSerdeProvider = null;
+            try {
+                metadataUpdateHandleSerdeProvider = connector.getConnectorMetadataUpdateHandleSerdeProvider();
+                requireNonNull(metadataUpdateHandleSerdeProvider, format("Connector %s returned null metadata update handle serde provider", connectorId));
+            }
+            catch (UnsupportedOperationException ignored) {
```

Comment:
This should never happen, so should we throw instead?

Reply:
This mimics rest of the connector provider behavior. See code above/below.